doc/cost-model-overview/cost-model-overview.tex

```diff
@@ -0,0 +1,386 @@
+        \verb|listData|                        & 1 & \const & \const \\
+        \verb|mapData|                         & 1 & \const & \const \\
+        \verb|mkCons|                          & 2 & \const & \const \\
+        \verb|mkNilData|                       & 1 & \const & \const \\
+        \verb|mkNilPairData|                   & 1 & \const & \const \\
+        \verb|mkPairData|                      & 2 & \const & \const \\
+        \verb|modInteger|                      & 2 & \quadrxy & \liny  \\
+        \verb|multiplyInteger|                 & 2 & \linmul & \linplus \\
+        \verb|nullList|                        & 1 & \const & \const \\
+        \verb|quotientInteger|                 & 2 & \quadrxy & \linsub \\
+        \verb|remainderInteger|                & 2 & \quadrxy & \liny \\
+        \verb|serialiseData|                   & 1 & \linx & \linx \\
+        \verb|sha2_256|                        & 1 & \linx & \const \\
+        \verb|sha3_256|                        & 1 & \linx & \const \\
+        \verb|sliceByteString|                 & 3 & \linz & \linz \\
+        \verb|sndPair|                         & 1 & \const & \const \\
+        \verb|subtractInteger|                 & 2 & \linmax & \linmax \\
+        \verb|tailList|                        & 1 & \const & \const \\
+        \verb|trace|                           & 2 & \const & \const \\
+        \verb|unBData|                         & 1 & \const & \const \\
+        \verb|unConstrData|                    & 1 & \const & \const \\
+        \verb|unIData|                         & 1 & \const & \const \\
+        \verb|unListData|                      & 1 & \const & \const \\
+        \verb|unMapData|                       & 1 & \const & \const \\
+        \verb|verifyEcdsaSecp256k1Signature|   & 3 & \const & \const \\
+        \verb|verifyEd25519Signature|          & 3 & \liny & \const \\
+        \verb|verifySchnorrSecp256k1Signature| & 3 & \liny & \const \\
+        \hline
+\end{tabular}
+    \caption{Costing functions for Plutus Core built-in functions (2)}
+    \label{table:costing-functions-2}
+\end{table}
+
+\paragraph{Note 1.}
+The CPU costing functions for the four integer division functions
+(\verb|divideInteger|, \verb|modInteger|, \verb|quotientInteger| and
+\verb|remainderInteger|) are rather complicated.  For $x<y$ the cost is constant
+since essentially no work has to be done; however for $x \geq y$ the cost is a
+quadratic function of the sizes of the two inputs with a lower bound to ensure
+that the cost is never negative:
+
+$$
+\text{CPU cost} = \begin{cases}
+  \const & \text{if $x<y$} \\
+  \max(d, c_{00} + c_{10}x + c_{01}y + c_{20}x^2 + c_{11}xy + c_{02}y^2) & \text{if $x \geq y$}.
+\end{cases}
+$$
+
+\subsection{Cost model parameters in the protocol parameters}
+\label{sec:cost-model-parameters}
+In the protocol parameters the cost model parameters take the form of a list of
```

Comment:
This sentence is not exactly correct, since in the "real" protocol parameters, there are no names, but just the coefficient values in a big array: https://github.com/perturbing/plutus-plonk-example/blob/1451b2a67b445089a579298195895915b76a523f/local-testnet/scripts/babbage/conway-babbage-test-genesis.json#L28

Reply:
> in the "real" protocol parameters, there are no names
I think that if you ask the node for the protocol parameters you get back a JSON object where all of the protocol parameters are tagged with their names.  At least that's what I got when I queried a node with cardano-cli a while ago.  I'll try that again and see if the same thing happens.  I think that the node maybe has different views of the protocol parameters, some with tags and some without, but the tags have always been removed from the cost model parameters when the node calls `mKEvaluationContext`.  This is all a bit unclear to me.

Reply:
For me too. If they do provide a JSON to the user, then disregard what I said